Products/CMFPlone/events.py

```diff
@@ -37,3 +42,38 @@ def removeBase(event):
     https://dev.plone.org/ticket/13705
     """
     event.request.response.base = None
+
+
+@adapter(IContentish, IObjectRemovedEvent)
+def handle_content_removal(obj, event):
+    """Event handler for content removal
+
+    This intercepts standard content removal and puts the item in the recycle bin
+    instead of letting it be deleted if the recycle bin is enabled.
+    """
+    # Ignore if the object is being moved
+    if getattr(obj, "_v_is_being_moved", False):
+        return
+
+    # Get the recycle bin
+    recycle_bin = queryUtility(IRecycleBin)
+    if recycle_bin is None or not recycle_bin.is_enabled():
+        return
+
+    # Only process if this is a direct deletion (not part of container deletion)
+    if event.newParent is not None:
+        return
+
+    # Get original information
+    original_container = event.oldParent
+    original_path = "/".join(obj.getPhysicalPath())
+
+    # Add to recycle bin
+    try:
+        recycle_bin.add_item(obj, original_container, original_path)
+    except Exception as e:
+        # Log but don't prevent deletion if recycle bin fails
+        import logging
+
+        logger = logging.getLogger("Products.CMFPlone.RecycleBin")
+        logger.exception(f"Error adding item to recycle bin: {e}")
```

Comment:
If the user is expecting the content to still exist in the recycle bin after it was deleted, then it's a bad surprise if it sometimes is not. I'd say we should let the exception get raised, so that it's obvious there's a problem that needs to be solved.

Reply:
done in https://github.com/plone/Products.CMFPlone/pull/4168/commits/01b908e11ef4a9171ab4fd40ee711b1892eeb759